File: 3451-string-compression-iii/3451-string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder sb = new StringBuilder(); 
        int count = 1; 
        char curr = word.charAt(0); 

        for(int i = 1; i < word.length(); i++){
            char compare = word.charAt(i); 

            if(curr == compare){
                count++; 
                if(count > 9){
                    sb.append(count - 1).append(curr); 
                    count = 1; 
                }
            } else{
                sb.append(count).append(curr); 
                curr = compare; 
                count = 1; 
            }
        }

        sb.append(count).append(curr); 

        return sb.toString();  
    }
}

File: 3451-string-compression-iii/3451-string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp = ""; 
        int i = 0; 
        while(i < word.length()){
            char curr = word[i]; 
            int count = 1; 
            while(i + 1 < word.length() && word[i+1] == curr && count < 9){
                count++; 
                i++; 
            }

            comp += (count + '0'); 
            comp += (curr); 
            i++; 
        }

        return comp; 
    }
};